pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -120,28 +116,38 @@ func (c *Controller) Start(stopCh <-chan struct{}) {
 		return
 	}
 
-	go wait.Until(c.sync, c.interval, stopCh)
+	// wait until first successfully sync
+	// this blocks apiserver startup so poll with a short interval
+	err := wait.PollUntilContextCancel(wait.ContextForChannel(stopCh), 100*time.Millisecond, true, func(ctx context.Context) (bool, error) {
```

Comment:
Why not just call `c.sync()` directly or:

```
for {
    err := c.sync()
    if err == nil {
        break
    }
    // log
}
```

Reply:
you have to honor the stopCh

Reply:
ehhhhhhh, ok.  Can you toss in a comment here saying something like that?

Reply:
done